python/lsst/ts/epm/snmp_data_client.py

```diff
@@ -0,0 +1,243 @@
+# This file is part of ts_epm.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the Vera Rubin Observatory
+# Project (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["SnmpDataClient"]
+
+import asyncio
+import logging
+import types
+import typing
+
+import yaml
+from lsst.ts import salobj
+from lsst.ts.ess import common
+from pysnmp.hlapi import (
+    CommunityData,
+    ContextData,
+    ObjectIdentity,
+    ObjectType,
+    SnmpEngine,
+    UdpTransportTarget,
+    nextCmd,
+)
+
+from .mib_tree import mib_tree
```

Comment:
Code runs during this import here. If something goes wrong it is unrecoverable, unless you protect this with a try/except (please don't).

Reply:
I reworked the code so this is all in a class now.